pagopa/io-app
# chore: [IOBP-933] Add exponential backoff retry into wallet section screen

## ⚠️ This PR depends on #6314 -> Please review the dependent PR before reviewing this ⚠️

## Short description
This PR introduces the management of exponential backoff retry with a banner on the main "Wallet" screen. The banner will appear if at least one of the three requests from the B&P domain (idpay, cgn, and payment methods) fails. When one of the requests fails, tapping the CTA to retry will trigger all the failed requests. The retry logic follows an exponential backoff, where the wait time increases exponentially with each retry attempt to a maximum of 3 minutes.

## List of changes proposed in this pull request
- Added a `WalletCardsCategoryRetryErrorBanner` component that handles all the logic when it should be rendered (only when at least one of the three requests has failed)
- Managed the 404 of the CGN card in a different way, if we receive 404 from the backend it shouldn't be tracked as an error but as a none value (it means that the user doesn't have a CGN);

## How to test
- With the app in dev env and the io-dev-server started, try to hardcode a 500 error response on the dev-server in one of the three requests that are emitted inside the Wallet home screen (CGN, Payments methods and IDPay) [Reminder: If you want test IDPay cards, please enable the feature flag `IDPay` inside the settings screen;
- If the banner is visible, double-check that it also has the exponential backoff retry

## Preview
In this Preview, I forced the CGN API to respond 500 error

https://github.com/user-attachments/assets/d8405b94-671d-4270-987c-4f3f35e6f21b
